### PR TITLE
Clippy cleanup

### PR DESCRIPTION
**Pull request by ark0f**
- Fix `try-runtime` feature cannot be compiled
- Check all features with clippy in CI
- Remove `gear check benchmark`

### CODE REVIEW

**Review by shamilsan — COMMENTED** [review]
Why benchmark has been removed?

**Comment by ark0f** [comment]
There was `runtime-benchmarks` feature which is redundant now because clippy checks all the features

File: pallets/gear/src/benchmarking/code.rs [hunk 1]
```diff
@@ -263,65 +262,6 @@ impl<T: Config> WasmModule<T>
 where
     T: Config,
 {
-    /// Uses the supplied wasm module and instruments it when requested.
-    pub fn instrumented(code: &[u8], inject_gas: bool, inject_stack: bool) -> Self {
```
Comment (shamilsan) [inline pallets/gear/src/benchmarking/code.rs:267]: I mean that bechmarks related code. Is it not need anymore?

Comment (ark0f) [reply pallets/gear/src/benchmarking/code.rs:267]: Yes, it's completely unused

**Review by shamilsan — APPROVED** [review]

**Review by breathx — APPROVED** [review]

**Pull request merged**